src/qmxgraph/decoration_contents.py

```diff
@@ -125,17 +131,17 @@ class TableData:
     :ivar optional[str] style: A inline style for the element.
     """
 
-    tag = attr.ib(default='td', init=False)
-    contents = attrib_table_contents(str, Image)
-    colspan = attr.ib(default=1, validator=_is_int)
-    rowspan = attr.ib(default=1, validator=_is_int)
-    style = attr.ib(default=None, validator=attr.validators.optional(_is_str))
+    tag: str = attr.ib(default='td', init=False)
+    contents: List[Union[str, Image]] = attrib_table_contents(str, Image)
+    colspan: int = attr.ib(default=1, validator=_is_int)
```

Comment:
Given that

```python
_is_int = attr.validators.instance_of(int)
_is_str = attr.validators.instance_of(str)
```
maybe those can be changed to things like

```python
colspan: int = 1
```

I don't know if there is a replacement to `init=False` using that syntax.

Reply:
Note that they are not equivalent:

```python
colspan: int = 1
```

Will not validate this *at runtime*, while this does:

```python
colspan: int = attr.ib(default=1, validator=_is_int)
```

When a project is sufficiently type-checked, I prefer to avoid the runtime checks and let the type checker catch those errors, but I would prefer to not change too much runtime at this point.

Reply:
I though attrs did add type checking to those.
I learned something =)